src/Adapter/MSTest.TestAdapter/MSTestExecutor.cs

```diff
@@ -86,6 +86,7 @@ public void RunTests(IEnumerable<string>? sources, IRunContext? runContext, IFra
         try
         {
             MSTestSettings.PopulateSettings(runContext);
+            MSTestSettings.ValidateSettings(frameworkHandle);
```

Comment:
Might be worth adding a comment in the other RunTests method to say why we don't validate there.